### PR TITLE
fix: should use server-entires error & modify ServerRoot as ReactElement

### PR DESCRIPTION
**Pull request by GiveMe-A-Name**
## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/runtime/plugin-runtime/src/core/server/requestHandler.ts [hunk 1]
```diff
@@ -25,7 +26,7 @@ export type HandleRequestOptions = Exclude<
 
 export type HandleRequest = (
   request: Request,
-  serverRoot: React.ComponentType, // App, routes,
+  ServerRoot: React.ReactElement, // App, routes,
```
Comment (zllkjc) [inline packages/runtime/plugin-runtime/src/core/server/requestHandler.ts:29]: the name for `ReactElement` type is better use lowercase, serverRoot is good

File: packages/runtime/plugin-runtime/src/core/server/requestHandler.ts [hunk 2]
```diff
@@ -213,7 +214,11 @@ export const createRequestHandler: CreateRequestHandler =
         `${CHUNK_CSS_PLACEHOLDER}</head>`,
       );
 
-      const response = await handleRequest(request, serverRoot, {
+      const ServerRoot = React.createElement(serverRoot, {
```
Comment (zllkjc) [inline packages/runtime/plugin-runtime/src/core/server/requestHandler.ts:217]: We should pass a variable capable of writing JSX syntax to the `requestHandler`, then user can write `<ServerRoot>` in their code. So, should it be written like this here:

```tsx
const ServerRoot = (props: any) => {
  return React.createElement(createRoot(), {
    ...props,
    _internal_context: Object.assign(context, { ssr: true }),
  });
};
```

File: packages/runtime/plugin-runtime/src/core/server/shared.ts [hunk 3]
```diff
@@ -7,13 +7,13 @@ export type SSRConfig = NonNullable<ServerUserConfig['ssr']>;
 
 export type RenderStreaming = (
   request: Request,
-  serverRoot: React.ComponentType<any>,
+  ServerRoot: React.ReactElement,
```
Comment (zllkjc) [inline packages/runtime/plugin-runtime/src/core/server/shared.ts:10]: also serverRoot is better

File: packages/runtime/plugin-runtime/src/core/server/string/index.ts
Comment (zllkjc) [inline packages/runtime/plugin-runtime/src/core/server/string/index.ts:1]: 这个文件也是，应该不用改造，renderString 的时候，拿到的是 React Element，用小写会好

**Comment by zllkjc** [comment]
I think we should modify the file `template.server.ts`, like: 

```tsx
const handleRequest = async (request, ServerRoot, options) => {

  const body = await #render(request, <ServerRoot />, options);

  return new Response(body, {
    headers: {
      'content-type': 'text/html; charset=utf-8',
      #headers
    },
  })
};

```

File: packages/runtime/plugin-runtime/src/core/server/requestHandler.ts (hunk 1 above)
Comment (GiveMe-A-Name) [reply packages/runtime/plugin-runtime/src/core/server/requestHandler.ts]: Done

File: packages/runtime/plugin-runtime/src/core/server/shared.ts (hunk 3 above)
Comment (GiveMe-A-Name) [reply packages/runtime/plugin-runtime/src/core/server/shared.ts]: Done~

File: packages/runtime/plugin-runtime/src/core/server/string/index.ts
Comment (GiveMe-A-Name) [reply packages/runtime/plugin-runtime/src/core/server/string/index.ts:1]: done~

File: packages/runtime/plugin-runtime/src/core/server/requestHandler.ts (hunk 2 above)
Comment (GiveMe-A-Name) [reply packages/runtime/plugin-runtime/src/core/server/requestHandler.ts]: Done~

**Review by zllkjc — APPROVED** [review]

**Pull request merged**